src/server/mod.rs

```diff
@@ -135,7 +134,7 @@ pub fn handle_client(mut stream: TcpStream,
 
                             // Send ServerHello message
                             let encoded: Vec<u8> = serialize(
-                                &Message::ServerHello(id, world_lock.clone()),
+                                &Message::ServerHello(id, (*world).clone()),
```

Comment:
Here again. And probably in some other places too :)

Reply:
Done.